--- fidosetup.orig.sh	2021-12-14 12:16:53.495850499 +0100
+++ fidosetup.sh	2021-12-14 12:35:05.242119962 +0100
@@ -305,6 +305,15 @@
 sed -i "s|@OTHER_DOMAINS@||g" ${TMP}/binkd.cfg
 
 echo ""
+echo "  * Creating Fidian log folder"
+mkdir -p /var/log/fidian
+
+echo ""
+echo "  * Fixing logfiles location for Docker"
+sed -i "s|/var/log/binkd|/var/log/fidian|g" ${TMP}/binkd.cfg
+sed -i "s|/var/log/husky|/var/log/fidian|g" ${TMP}/config
+
+echo ""
 echo "  * Copying configuration files to their destinations"
 
 for f in ${CFGFILES}
@@ -353,8 +362,6 @@
   echo "*/1 * * * *   root  /usr/local/sbin/semacheck.sh" >/etc/cron.d/fido
   echo "*/10 * * * *  ftn   touch /var/spool/ftn/flags/poll; touch /var/spool/ftn/flags/echoscan; touch /var/spool/ftn/flags/netscan" >>/etc/cron.d/fido
   echo '8 6 * * 3     ftn   /usr/bin/cronic /usr/bin/sqpack -c /etc/husky/config "*"; chmod -R 770 /var/spool/ftn/msgbase; find /var/spool/ftn/msgbase -type f -exec chmod 660 "{}" \;' >>/etc/cron.d/fido
-  echo "    - restarting cron"
-  /usr/sbin/service cron restart
 fi
 
 echo ""
@@ -436,6 +443,11 @@
 sudo -u fido gnlnx -Q -F /home/fido/.golded/golded.cfg
 
 echo ""
+echo "  * Patching Fidian scripts for Docker"
+wget --quiet -O "${TMP}/docker.patch" https://raw.githubusercontent.com/stimpy23/fidian-docker/master/extras/docker.patch
+patch -s "${TMP}/scripts_fidian-master/fidosetup.sh" "${TMP}/docker.patch"
+
+echo ""
 if  [ ! -e /usr/local/sbin/fidoconfig.sh ] || 
     [ ! -e /usr/local/sbin/fidosetup.sh ] || 
     [ ! -e /usr/local/sbin/fidouninstall.sh ]
@@ -503,12 +515,6 @@
 locale-gen
 
 echo ""
-echo "  * (Re-)starting binkd"
-/usr/sbin/service binkd stop
-sleep 5
-/usr/sbin/service binkd start
-
-echo ""
 echo "  * Cleaning up temp files"
 rm -rf ${TMP}
 
