--- fidosetup.orig.sh	2021-12-14 16:32:31.212191434 +0100
+++ fidosetup.sh	2021-12-14 16:35:15.765878114 +0100
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-CFGFILES="/etc/husky/areas /etc/binkd/binkd.cfg /etc/binkd/binkd.inc /home/fido/.golded/charsets.cfg /etc/husky/config /etc/husky/fileareas /home/fido/.golded/golded.cfg /home/fido/.golded/local.cfg /home/fido/.golded/goldkeys.cfg /etc/husky/links /etc/husky/route /usr/local/sbin/poll.sh /usr/local/sbin/toss.sh /usr/local/sbin/tick.sh /usr/local/sbin/echoscan.sh /usr/local/sbin/netscan.sh /usr/local/sbin/fidocomplete.sh /etc/logrotate.d/husky /usr/local/sbin/semacheck.sh /usr/local/sbin/fidoupdate.sh"
+CFGFILES="/etc/husky/areas /etc/binkd/binkd.cfg /etc/binkd/binkd.inc /home/fido/.golded/charsets.cfg /etc/husky/config /etc/husky/fileareas /home/fido/.golded/golded.cfg /home/fido/.golded/local.cfg /home/fido/.golded/goldkeys.cfg /etc/husky/links /etc/husky/route /usr/local/sbin/poll.sh /usr/local/sbin/toss.sh /usr/local/sbin/tick.sh /usr/local/sbin/echoscan.sh /usr/local/sbin/netscan.sh /usr/local/sbin/fidocomplete.sh /usr/local/sbin/semacheck.sh /usr/local/sbin/fidoupdate.sh"
 VERSION="0.2.4"
 declare -A CFG
 
@@ -305,6 +305,17 @@
 sed -i "s|@OTHER_DOMAINS@||g" ${TMP}/binkd.cfg
 
 echo ""
+echo "  * Creating Fidian-docker log folder"
+mkdir -p /var/log/fidian
+chown root:ftn /var/log/fidian
+chmod 775 /var/log/fidian
+
+echo ""
+echo "  * Fixing logfiles location for Docker"
+sed -i "s|/var/log/binkd|/var/log/fidian|g" ${TMP}/binkd.cfg
+sed -i "s|/var/log/husky|/var/log/fidian|g" ${TMP}/config
+
+echo ""
 echo "  * Copying configuration files to their destinations"
 
 for f in ${CFGFILES}
@@ -327,8 +338,6 @@
 mkdir -p /var/spool/ftn/transit
 mkdir -p /var/spool/ftn/flags
 mkdir -p /var/spool/ftn/msgbase/fido
-mkdir -p /var/log/husky
-mkdir -p /var/log/binkd
 touch /var/spool/ftn/msgbase/netmail.jdt
 touch /var/spool/ftn/msgbase/netmail.jdx
 touch /var/spool/ftn/msgbase/netmail.jhr
@@ -353,8 +362,6 @@
   echo "*/1 * * * *   root  /usr/local/sbin/semacheck.sh" >/etc/cron.d/fido
   echo "*/10 * * * *  ftn   touch /var/spool/ftn/flags/poll; touch /var/spool/ftn/flags/echoscan; touch /var/spool/ftn/flags/netscan" >>/etc/cron.d/fido
   echo '8 6 * * 3     ftn   /usr/bin/cronic /usr/bin/sqpack -c /etc/husky/config "*"; chmod -R 770 /var/spool/ftn/msgbase; find /var/spool/ftn/msgbase -type f -exec chmod 660 "{}" \;' >>/etc/cron.d/fido
-  echo "    - restarting cron"
-  /usr/sbin/service cron restart
 fi
 
 echo ""
@@ -436,6 +443,15 @@
 sudo -u fido gnlnx -Q -F /home/fido/.golded/golded.cfg
 
 echo ""
+echo "  * Patching Fidian scripts for Docker"
+patchurl="https://raw.githubusercontent.com/stimpy23/fidian-docker/master/extras/docker.patch"
+if [ "${BRANCH}" == "dev" ]; then
+  patchurl="https://gitlab.ambhost.net/stimpy/docker_fidian/-/raw/master/extras/docker.patch"
+fi
+wget --quiet -O "${TMP}/docker.patch" ${patchurl}
+patch -s "${TMP}/scripts_fidian-master/fidosetup.sh" "${TMP}/docker.patch"
+
+echo ""
 if  [ ! -e /usr/local/sbin/fidoconfig.sh ] || 
     [ ! -e /usr/local/sbin/fidosetup.sh ] || 
     [ ! -e /usr/local/sbin/fidouninstall.sh ]
@@ -503,12 +519,6 @@
 locale-gen
 
 echo ""
-echo "  * (Re-)starting binkd"
-/usr/sbin/service binkd stop
-sleep 5
-/usr/sbin/service binkd start
-
-echo ""
 echo "  * Cleaning up temp files"
 rm -rf ${TMP}
 
